nucliadb_vectors2/src/data_point_provider/writer.rs

```diff
@@ -245,10 +245,26 @@ impl Writer {
         keep.push(new_online_data_point);
         self.online_data_points = keep;
 
+        let mut new_delete_log = self.delete_log.clone();
         if let Some(oldest_age) = self.online_data_points.iter().map(|dp| dp.journal.time()).reduce(std::cmp::min) {
-            self.delete_log.prune(oldest_age);
+            new_delete_log.prune(oldest_age);
         }
 
+        let state = State {
+            delete_log: new_delete_log,
+            data_point_list: self.online_data_points.iter().map(|dp| dp.journal.id()).collect(),
```

Comment:
you are iterating `self.online_data_points` twice, once for obtaining the oldest age and here. Not a big deal if we keep the number of data points low, but maybe we could organice the code in a way that only one iteration suffice?